packages/frontend-2/components/settings/workspaces/members/Table.vue

```diff
@@ -0,0 +1,181 @@
+    v-model:open="showChangeUserRoleDialog"
+    :name="userToModify?.name ?? ''"
+    :old-role="oldRole"
+    :new-role="newRole"
+    @update-role="onUpdateRole"
+  />
+</template>
+
+<script setup lang="ts">
+// Todo: Enable searching once supported
+import type { WorkspaceRoles } from '@speckle/shared'
+// import { MagnifyingGlassIcon } from '@heroicons/vue/24/outline'
+// import { useDebouncedTextInput } from '@speckle/ui-components'
+import { settingsWorkspacesMembersQuery } from '~~/lib/settings/graphql/queries'
+import { workspaceUpdateRoleMutation } from '~~/lib/workspaces/graphql/mutations'
+import { useActiveUser } from '~~/lib/auth/composables/activeUser'
+import { useMutation, useQuery } from '@vue/apollo-composable'
+import { useGlobalToast, ToastNotificationType } from '~~/lib/common/composables/toast'
+import {
+  convertThrowIntoFetchResult,
+  getCacheId,
+  getFirstErrorMessage
+} from '~~/lib/common/helpers/graphql'
+import type { SettingsWorkspacesMembersQuery } from '~~/lib/common/generated/gql/graphql'
+
+type UserItem = {
+  id: string
+  role: string
+  name: string
+  verified?: boolean | null
+  avatar?: string | null
+  company?: string | null
+}
+
+const props = defineProps<{
+  workspaceId: string
+}>()
+
+const { triggerNotification } = useGlobalToast()
+// const { on, bind, value: search } = useDebouncedTextInput()
+const { activeUser } = useActiveUser()
+const { mutate: updateChangeRole } = useMutation(workspaceUpdateRoleMutation)
+
+const { result } = useQuery<SettingsWorkspacesMembersQuery>(
+  settingsWorkspacesMembersQuery,
+  () => ({
+    workspaceId: props.workspaceId
+  })
+)
+
+const showChangeUserRoleDialog = ref(false)
+const newRole = ref<WorkspaceRoles>()
+const userToModify = ref<UserItem>()
+
+const members = computed(() => {
+  if (result?.value?.workspace?.team) {
+    return result.value.workspace.team.map(({ user, ...rest }) => ({
+      ...user,
+      ...rest
+    }))
+  }
+
+  return []
+})
+
+const oldRole = computed(() => userToModify.value?.role as WorkspaceRoles)
+const isCurrentUser = (id: string) => id === activeUser.value?.id
+
+const openChangeUserRoleDialog = (
+  user: UserItem,
+  newRoleValue?: WorkspaceRoles | WorkspaceRoles[]
+) => {
+  if (!newRoleValue) return
+  userToModify.value = user
+  newRole.value = Array.isArray(newRoleValue) ? newRoleValue[0] : newRoleValue
+  showChangeUserRoleDialog.value = true
+}
+
+const onUpdateRole = async () => {
+  if (!userToModify.value || !newRole.value) return
+
+  const mutationResult = await updateChangeRole(
+    {
+      input: {
+        userId: userToModify.value.id,
+        role: newRole.value,
+        workspaceId: props.workspaceId
+      }
+    },
+    {
+      update: (cache, { data }) => {
+        const userId = userToModify.value?.id
+        const newRoleVal = newRole
+
+        if (data?.workspaceMutations?.updateRole) {
+          cache.modify({
+            id: getCacheId('WorkspaceCollaborator', userId as string),
```

Comment:
1. Instead of overriding the type, store `userId` already in the scope where you know it exists - right after the check in line 140. The update() callback is invoked asynchronously and theoretically it is possible for userToModify to not have the same value then anymore.
2. This whole thing could be simpler without the need to do custom cache updates if you'd just ask back for the updated `Workspace.team` field in the mutation https://www.apollographql.com/docs/react/data/mutations/#include-modified-objects-in-mutation-responses